waiter/src/waiter/main.clj

```diff
@@ -77,14 +77,15 @@
    :state core/state
    :http-server (pc/fnk [[:routines generate-log-url-fn waiter-request?-fn websocket-request-acceptor]
                          [:settings cors-config host port server-options support-info websocket-config]
-                         [:state cors-validator router-id]
+                         [:state cors-validator router-id server-name]
                          handlers] ; Insist that all systems are running before we start server
                   (let [options (merge (cond-> server-options
                                          (:ssl-port server-options) (assoc :ssl? true))
                                        websocket-config
                                        {:ring-handler (-> (core/ring-handler-factory waiter-request?-fn handlers)
                                                           (cors/wrap-cors-preflight cors-validator (:max-age cors-config))
                                                           core/wrap-error-handling
+                                                          (core/wrap-grpc-status server-name)
```

Comment:
Should we install the middleware around the `process-request` handler only?  I can't think of any other path where gRPC statuses need to be generated.

Reply:
The authentication responses are not generated by `process-request`